### ISSUE TITLE
Certain options don't animate back

### ISSUE THREAD

**Issue by Andre-Pham** [Andre-Pham/Yonder#68]
Notice how the travel option doesn't animate back:

https://github.com/Andre-Pham/Yonder/assets/60597159/6bd2cf50-4b39-4f32-ae98-ddc6bff1bcd5

This behaviour is VERY inconsistent.

In this case, I'm testing the loot button. When I click this test button (which adds a loot bag to the player), the loot option doesn't animate in:

https://github.com/Andre-Pham/Yonder/assets/60597159/ca07fbbf-9647-4c77-8762-b8b3b245c2bf

Through testing the loot option bug, here are my observations:
* It still occurs even without the horizontal scroll button
* It can still occur when horizontal scrolling is active
* It still occurs when removing the `showScrollDirectionToggle` 
* It still occurs when removing all action states outside of the animation
* It still occurs even when I manually add the `transition` modifier to all grid options
* It ONLY occurs on the related state - if I finish looting, it triggers on only the loot button

**Comment by Andre-Pham** [comment]
Further observations:
* The animation bug doesn't occur if the condition for showing the option is replaced with "true" (the view is always showing)
* The animation bug doesn't occur if you add a delay to `closeActions`

My theory: there's a slight delay in calculating if the option should be shown, which means it isn't included in the animation batch

**Comment by Andre-Pham** [comment]
Solution: receive the published state on the main thread.

```swift
self.playerViewModel.$lootBagViewModel
    .receive(on: DispatchQueue.main)
    .sink { [weak self] _ in
        guard let self = self else { return }
        print("Loot option active: \(self.lootOptionActive)")
        self.closeActions()
    }
    .store(in: &self.subscriptions)
```

**Comment by Andre-Pham** [comment]
Fixed: 2f96e7f356df13eef949ab078fdb4790c7149f1e
